src/main/java/com/app/onlinebookstore/service/impl/CartItemServiceImpl.java

```diff
@@ -0,0 +1,85 @@
+package com.app.onlinebookstore.service.impl;
+
+import com.app.onlinebookstore.dto.CartItemDto;
+import com.app.onlinebookstore.dto.CreateCartItemRequestDto;
+import com.app.onlinebookstore.dto.UpdateCartItemDto;
+import com.app.onlinebookstore.exaption.EntityNotFoundException;
+import com.app.onlinebookstore.mapper.CartItemMapper;
+import com.app.onlinebookstore.model.CartItem;
+import com.app.onlinebookstore.model.ShoppingCart;
+import com.app.onlinebookstore.model.User;
+import com.app.onlinebookstore.repository.BookRepository;
+import com.app.onlinebookstore.repository.CartItemRepository;
+import com.app.onlinebookstore.repository.ShoppingCartRepository;
+import com.app.onlinebookstore.repository.UserRepository;
+import com.app.onlinebookstore.service.CartItemService;
+import com.app.onlinebookstore.service.UserService;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class CartItemServiceImpl implements CartItemService {
+    private final BookRepository bookRepository;
+    private final UserService userService;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemMapper cartItemMapper;
+    private final CartItemRepository cartItemRepository;
+    private final UserRepository userRepository;
+
+    @Override
+    @Transactional
+    public CartItemDto save(CreateCartItemRequestDto requestDto) {
+        CartItem cartItem = new CartItem();
+        cartItem.setBook(bookRepository.findById(requestDto.getBookId()).get());
+        cartItem.setQuantity(requestDto.getQuantity());
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        Optional<User> user = userRepository.findByEmail(authentication.getName());
+        Long userId = user.get().getId();
+        setShoppingCartAndCartItems(shoppingCartRepository
+                .findByUserId(userId).get().getId(), cartItem);
+        return cartItemMapper.toDto(cartItemRepository.save(cartItem));
+    }
+
+    @Override

```

Comment:
```suggestion
    @Override
    @Transactional
```
This method also should be transactional, as you fetching entity, setting a new field and saving it